merkle-tree/concurrent/tests/tests.rs

```diff
@@ -634,49 +636,125 @@ where
     }
 }
 
+fn compat_canopy<H, const HEIGHT: usize>()
```

Comment:
do you think it could make sense to add a long running test that tries height 26 with canopy 10, to explicitly test the parameters we are going to use onchain?
Another interesting testcase could be comparing a tree of height 10 and canopy zero with a  tree of height 10 with canopy 10 and fill it up to assert that the canopy behaves exactly the same for every leaf.

Reply:
Can you think of other edge cases to test?

Reply:
I can give it more thought, but the only other scenario which comes immediately to my mind is testing indexed MT with canopy (appends followed by update).